pkg/controllers/management/gke/gke_cluster_handler.go

```diff
@@ -418,37 +418,45 @@ func (e *gkeOperatorController) deployGKEOperator() error {
 }
 
 func (e *gkeOperatorController) generateSATokenWithPublicAPI(cluster *mgmtv3.Cluster) (string, *bool, error) {
-	ctx := context.Background()
-	ts, err := controller.GetTokenSource(ctx, e.SecretsCache, cluster.Spec.GKEConfig)
+	restConfig, err := e.getRestConfig(cluster, (&net.Dialer{Timeout: 30 * time.Second, KeepAlive: 30 * time.Second}).DialContext)
 	if err != nil {
 		return "", nil, err
 	}
-
-	netDialer := net.Dialer{
-		Timeout:   30 * time.Second,
-		KeepAlive: 30 * time.Second,
-	}
 	publicAccess := new(bool)
-	serviceToken, err := generateSAToken(cluster.Status.APIEndpoint, cluster.Status.CACert, ts, netDialer.DialContext)
+	serviceToken, err := clusteroperator.GenerateSAToken(restConfig)
 	if err != nil {
+		*publicAccess = true
```

Comment:
Doesn't this actually mean publicAccess should be false? Should this variable be renamed to `mustTunnel`?

Reply:
Yes, it should be `mustTunnel`. Will fix.